src/dns/EtcHosts.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "base/CharacterSet.h"
+#include "configuration/File.h"
+#include "dns/EtcHosts.h"
+#include "fqdncache.h"
+#include "ipcache.h"
+#include "parser/Tokenizer.h"
+#include "SquidConfig.h"
+
+DefineRunnerRegistratorIn(Dns, EtcHosts);
+
+SBuf Dns::EtcHosts::Path;
+
+void
+Dns::EtcHosts::parse()
+{
+    if (Path.isEmpty() || Path.cmp("none") == 0)
+        return;
+
+    etcHostsFile = new Configuration::File(Path.c_str());
```

Comment:
Please do not use a raw pointer to store heap-allocated objects.

AFAICT, the proposed Dns::EtcHosts::etcHostsFile data member should be converted into an ordinary local non-pointer variable instead, eliminating this problem and simplifying code.